### ISSUE TITLE
False positive: duplicate condition in if branch

### ISSUE THREAD

**Issue by BenMorel** [kalessil/phpinspectionsea#381]
When testing twice the same variable in nested `if`s, even if the variable has been overwritten, the inspection reports:

> The condition is duplicated in another if/elseif branch.

How to reproduce:

```php
function getUserIdFromSession() { return null; }
function getUserIdFromPermanentSession() { return null; }

function test()
{
    $userId = getUserIdFromSession();

    if ($userId === null) {
        $userId = getUserIdFromPermanentSession();

        if ($userId === null) { // "The condition is duplicated in another if/elseif branch."
            return null;
        }

        // ...
    }
    
    // ...
}
```

It would be nice to detect when the variable has been overwritten!

**Comment by kalessil** [comment]
Fixed.

Note: further extension of #306

**Comment by BenMorel** [comment]
Thanks, that was fast!

**Comment by kalessil** [comment]
You are welcome =)

**Comment by kalessil** [comment]
Fixed #550, coming into 2.3.12
